src/tools/tests/storage-key-recipe-resolver-test.ts

```diff
@@ -393,5 +392,33 @@ describe('recipe2plan', () => {
           'No matching handles found for data.'
         );
     });
+    it('fails to resolve when user maps to a volatile create handle', Flags.withDefaultReferenceMode(async () => {
+      const manifest = await Manifest.parse(`\
+    particle Reader
+      data: reads Thing {name: Text}
+
+    particle Writer
+       data: writes Thing {name: Text}
+    
+    @trigger
+      launch startup
+      arcId writeArcId
```

Comment:
just as a side note (no change needed in this PR), this behavior is temporary because lack of `persistent` capability maps directly to `volatile` in the current implementation.
we would change it soon, so that if a recipe is mapping the store, it would become `ramdisk`, but not `volatile` (basically allow mapping of an in-memory store).
just an fyi

Reply:
Ah, thanks for that context. Yeah, that would also address the underlying issue here. 
The change here is easily reversible, when we're ready to make the switch.